### DISCUSSION TITLE
Possiblity to Change .xacro file

### DISCUSSION THREAD

**Discussion by lucasmartins131** [ctu-mrs/mrs_uav_system#142] [Q&A]
I want to use the MRS_UAV_SYSTEM to simulate a UAV with an arm in the base. 

Do you know if it is possible to change the .xacro file of the UAVs to create this arm ?

**Comment by dansmrcka** [comment]
Hi, sure you can. Models definitions are stored here: [https://github.com/ctu-mrs/mrs_simulation/tree/master/models/mrs_robots_description/urdf](https://github.com/ctu-mrs/mrs_simulation/tree/master/models/mrs_robots_description/urdf). You can expand any UAV model you like. There are many extensions to our models already defined, there is probably no official list, but you can find them here: [https://github.com/ctu-mrs/mrs_simulation/blob/master/config/spawner_params.yaml](https://github.com/ctu-mrs/mrs_simulation/blob/master/config/spawner_params.yaml). You can similarly add your arm model and then spawn the model with arguments you want, e.g. `rosservice call /mrs_drone_spawner/spawn "${UAV_NUMBERS[1]} $UAV_TYPE --enable-rangefinder --enable-bluefox-camera"`.

**Reply by lucasmartins131** [reply]
Thanks for the answer. I had some problems with installation, that's why I took long for reply, but now I think it is solved. 
After packages are installed do you know the typical path of the models ? I tried to find so I could change the model that is running in the examples just to check, but couldn't find it. 
Thank you again

**Reply by lucasmartins131** [reply]
Folder Found I think so : 
~/mrs_workspace/src/simulation/ros_packages/mrs_simulation/models/mrs_robots_description

**Comment by lucasmartins131** [comment]
Do you know if after changing the .xacro file it is necessary to build the ws ?

**Reply by klaxalk** [reply]
You do not need to recompile after changing the xacro in mrs_simulation.

**Comment by lucasmartins131** [comment]
Hello Guys 
I don't know if you still can help me from here but I have a doubt regarding changing the parameters: 

I only found one place for mass parameters and one place for body inertia and a motors inertia. 
For modeling an arm I would need to add a mass for it (that is not in CG) and also a inertia moment. 
Do you have any idea how to modify the .xacro file for modeling it ? 

Thank's again.

**Reply by dansmrcka** [reply]
Did you find a solution? I am not sure what you mean by CG, but you can define additional parameters and functions to fit your arm shape and mass. We are simplifying the inertia of a drone to be a cylinder of the drone's radius and height.
